crates/ruff_diagnostics/src/fix.rs

```diff
@@ -0,0 +1,43 @@
+use rustpython_parser::ast::Location;
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+use crate::edit::Edit;
+
+/// A collection of edits to be applied to a source file.
+#[derive(Default, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct Fix {
+    edits: Vec<Edit>,
+}
+
+impl Fix {
+    pub fn new(edits: Vec<Edit>) -> Self {
+        // TODO(charlie): Ensure that the edits are sorted and non-overlapping.
+        debug_assert!(!edits.is_empty(), "Fix must have at least one edit");
+
+        Self { edits }
+    }
+
+    pub fn none() -> Self {
+        Self::default()
+    }
+
+    pub fn is_none(&self) -> bool {
+        self.edits.is_empty()
+    }
```

Comment:
Nit: I would prefer `Fix::is_empty()` and `Fix::empty()`
```suggestion
    pub const fn empty() -> Self {
        Self::default()
    }

    pub const fn is_empty(&self) -> bool {
        self.edits.is_empty()
    }
```